src/ext/java/org/opentripplanner/ext/fares/impl/OrcaFareServiceImpl.java

```diff
@@ -0,0 +1,665 @@
+        addFareComponent(leg, fare, fareType, currency, legFare, 0);
+        cost += legFare;
+      }
+    }
+    cost += orcaFareDiscount;
+    if (cost < Float.POSITIVE_INFINITY) {
+      fare.addFare(fareType, getMoney(currency, cost));
+    }
+    return cost < Float.POSITIVE_INFINITY;
+  }
+
+  /**
+   * Adds a fare component to a given ride.
+   * @param leg Ride receiving fare component
+   * @param fareType Fare type for fare component
+   * @param totalFare Cost of leg fare after transfer
+   * @param transferDiscount Transfer discount applied or 0 if no transfer was used.
+   */
+  private static void addFareComponent(
+    Leg leg,
+    ItineraryFares itineraryFares,
+    FareType fareType,
+    Currency currency,
+    float totalFare,
+    float transferDiscount
+  ) {
+    var id = new FeedScopedId("orcaFares", "farePayment");
+    var riderCategory = new RiderCategory("orcaFares", getFareCategory(fareType), "");
+    var fareContainer = new FareContainer("orcaFares", usesOrca(fareType) ? "electronic" : "cash");
+    var duration = Duration.ZERO;
+    var money = new Money(currency, (int) (totalFare * 100));
+    var fareProduct = new FareProduct(
+      id,
+      "rideCost",
+      money,
+      duration,
+      riderCategory,
+      fareContainer
+    );
+    itineraryFares.addFareProduct(leg, fareProduct);
+    // If a transfer was used, then also add a transfer fare product.
+    if (transferDiscount > 0) {
+      var transferDiscountMoney = new Money(currency, (int) (transferDiscount * 100));
+      var transferFareProduct = new FareProduct(
+        id,
+        "transfer",
+        transferDiscountMoney,
+        duration,
+        riderCategory,
+        fareContainer
+      );
+      itineraryFares.addFareProduct(leg, transferFareProduct);
+    }
+  }
+
+  /**
+   * Check if trip falls within the transfer time window.
+   * @param freeTransferStartTime
+   * @param currentLegStartTime
+   */
+  private boolean inFreeTransferWindow(
+    ZonedDateTime freeTransferStartTime,
+    ZonedDateTime currentLegStartTime
+  ) {
+    // If there is no free transfer, then return false.
+    if (freeTransferStartTime == null) return false;
+    Duration duration = Duration.between(freeTransferStartTime, currentLegStartTime);
+    return duration.compareTo(MAX_TRANSFER_DISCOUNT_DURATION) < 0;
+  }
+
+  /**
+   * A free transfer can be applied if using Orca and the transit agency permits free transfers.
+   */
+  private boolean hasFreeTransfers(FareType fareType, RideType rideType) {
+    // King County Metro allows transfers on cash fare
+    return (
+      (permitsFreeTransfers(rideType) && usesOrca(fareType)) ||
+      (rideType == RideType.KC_METRO && !usesOrca(fareType))
+    );
+  }
+
+  /**
+   * All transit agencies permit free transfers, apart from these.
+   */
+  private boolean permitsFreeTransfers(RideType rideType) {
+    return rideType != RideType.WASHINGTON_STATE_FERRIES && rideType != RideType.SKAGIT_TRANSIT;
```

Comment:
```suggestion
    return switch (rideType) {
      case WASHINGTON_STATE_FERRIES, SKAGIT_TRANSIT -> false;
      default -> true;
    };
```

I find this a tiny bit more readable. If you like it you can apply it but it's kinda up to personal preference so no must.